File: src/widgets/AffirmationWidget.js
import React from 'react';
import { Text, VStack, Spacer } from '@expo/ui/swift-ui';
import { font, foregroundStyle, background, padding, frame } from '@expo/ui/swift-ui/modifiers';

const AffirmationWidget = (props) => {
  const { text, color1, color2 } = props;
  const primaryColor = color1 || '#A855F7';
  const secondaryColor = color2 || '#EC4899';

  return (
    <VStack
      alignment="center"
      spacing={8}
      modifiers={[
        frame({ maxWidth: 10000, maxHeight: 10000 }),
        background(primaryColor),
        padding({ all: 20 }),
      ]}
    >
      <Text
        modifiers={[
          font({ size: 10, weight: 'medium', design: 'rounded' }),
          foregroundStyle(secondaryColor),
        ]}
      >
        mirrorcle
      </Text>
      <Spacer />
      <Text
        modifiers={[
          font({ size: 18, weight: 'bold', design: 'rounded' }),
          foregroundStyle('#FFFFFF'),
          padding({ horizontal: 4 }),
        ]}
      >
        {text || 'I am worthy of love and respect'}
      </Text>
      <Spacer />
    </VStack>
  );
};

export default AffirmationWidget;

File: ios/ExpoWidgetsTarget/AffirmationWidget.swift
import WidgetKit
import SwiftUI
internal import ExpoWidgets

struct AffirmationWidget: Widget {
  let name: String = "AffirmationWidget"

  var body: some WidgetConfiguration {
    StaticConfiguration(kind: name, provider: WidgetsTimelineProvider(name: name)) { entry in
      if let node = entry.node {
        WidgetsDynamicView(source: entry.source, kind: .widget, node: node)
          .containerBackground(for: .widget) {
            Color(red: 168/255, green: 85/255, blue: 247/255)
          }
      } else {
        AffirmationWidgetPlaceholder()
          .containerBackground(for: .widget) {
            Color(red: 168/255, green: 85/255, blue: 247/255)
          }
      }
    }
    .configurationDisplayName("Daily Affirmation")
    .description("A new affirmation to inspire you every time you check your phone")
    .supportedFamilies([.systemSmall, .systemMedium, .systemLarge])
    .contentMarginsDisabled()
  }
}

struct AffirmationWidgetPlaceholder: View {
  var body: some View {
    VStack(spacing: 8) {
      Text("mirrorcle")
        .font(.system(size: 10, weight: .medium, design: .rounded))
        .foregroundColor(Color(red: 236/255, green: 72/255, blue: 153/255))
      Spacer()
      Text("I am worthy of love and respect")
        .font(.system(size: 18, weight: .bold, design: .rounded))
        .foregroundColor(.white)
        .multilineTextAlignment(.center)
        .padding(.horizontal, 4)
      Spacer()
    }
    .frame(maxWidth: .infinity, maxHeight: .infinity)
    .padding(20)
    .background(Color(red: 168/255, green: 85/255, blue: 247/255))
  }
}
